### DISCUSSION TITLE
xdist module in pytest is not recognized

### DISCUSSION THREAD

**Discussion by SalvaVamsiKasyap** [pytest-dev/pytest#10834] [Q&A]
PS C:\Users\Lenovo\PycharmProjects\Personal_Framework\Pytest_Sessions> pytest .\test_fixture_params.py -v -s -n 2 --html=test_fixtures_params.html     
ERROR: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: unrecognized arguments: -n 2
  inifile: None
  rootdir: C:\Users\Lenovo\PycharmProjects\Personal_Framework\Pytest_Sessions
  
  Can any body help me with this?

**Comment by RonnyPfannschmidt** [comment]
as no details on your environment and how you get there is provided,
the only answer i can provide is that the dice i rolled landed on `4`

**Reply by nicoddemus** [reply]
@RonnyPfannschmidt not sure being snarky like that is warranted here; I assume this person is new to Python/programming and does not understand the error (probably copied/pasted it from somewhere on the web).

@SalvaVamsiKasyap seems like you do not have the `pytest-xdist` plugin installed. From the command-line you will also need the `pytest-html` plugin (due to the `--html` flag). You can install them with `pip install pytest-xdist pytest-html`. See [How to install and use plugins](https://docs.pytest.org/en/stable/how-to/plugins.html) in the documentation.

**Comment by SalvaVamsiKasyap** [comment]
Hi @nicoddemus and @RonnyPfannschmidt ,

PS C:\Users\Lenovo\PycharmProjects\Personal_Framework\Pytest_Sessions> pip list
Package         Version
--------------- --------
attrs           22.2.0
colorama        0.4.6
execnet         1.9.0
iniconfig       2.0.0
numpy           1.24.2
packaging       23.0
pip             22.3
pluggy          1.0.0
psutil          5.9.4
py              1.11.0
pytest          7.2.2
pytest-html     3.2.0
pytest-metadata 2.0.4
**pytest-xdist    3.2.1**
pytz            2022.7.1
robotframework  6.0.2
setuptools      67.6.0
six             1.16.0
softest         1.2.0.0

I  have installed it actually xdist plugin i am getting the above error only when i am trying to run the test cases parllely

PS C:\Users\Lenovo\PycharmProjects\Personal_Framework\Pytest_Sessions> pip --version
pip 22.3 from C:\Python311\Lib\site-packages\pip (python 3.11)
PS C:\Users\Lenovo\PycharmProjects\Personal_Framework\Pytest_Sessions> pytest --version
pytest 7.2.0

**Reply by RonnyPfannschmidt** [reply]
Please check the paths of pip and pytest, in some cases they differ and can lead to confusion

Are you using virtualenv

How are your python installations managed

**Reply by nicoddemus** [reply]
> Please check the paths of pip and pytest, in some cases they differ and can lead to confusion

@SalvaVamsiKasyap you can use `where pip` and `where pytest` for that on Windows.
